crates/autopilot/src/run_loop.rs

```diff
@@ -405,20 +409,64 @@ impl RunLoop {
 
     /// Execute the solver's solution. Returns Ok when the corresponding
     /// transaction has been mined.
-    async fn settle(&self, driver: &infra::Driver, solved: &Solution) -> Result<(), SettleError> {
+    async fn settle(
+        &self,
+        driver: infra::Driver,
+        solved: &Solution,
+        auction_id: i64,
+    ) -> Result<(), SettleError> {
         let order_ids = solved.order_ids().copied().collect();
         self.persistence
             .store_order_events(order_ids, OrderEventLabel::Executing);
 
+        let eth = self.eth.clone();
+        let persistence = self.persistence.clone();
+        let max_blocks_wait = self.max_blocks_wait;
+        let (cancellation_tx, cancellation_rx) = watch::channel(false);
+        let settlement_tx_wait_task = tokio::spawn(async move {
+            let tag = auction_id.to_be_bytes();
```

Comment:
According to the https://github.com/cowprotocol/services/blob/aa5f4488360df5ea33f73785e65248c2c278470e/crates/driver/src/boundary/settlement.rs#L185